### DISCUSSION TITLE
How to run with AMD

### DISCUSSION THREAD

**Discussion by dmarsh400** [DrewThomasson/VoxNovel#28] [Q&A]
How can I make VoxNovel run with AMD either integrated or discreet GPU?

**Comment by DrewThomasson** [comment]
Sorry,as far as I can tell there's so way so far for that,

-Related ref
https://github.com/DrewThomasson/VoxNovel/issues/9

I'll look into it and update you on anything I can find though,


## The things we'd need to get working on AMD would be:

-BOOKNLP for book preprocessing
-XTTS for the text to speech
-styleTTS another option for text to speech

**Reply by dmarsh400** [reply]
I was just looking around and I see there are CUDA substitutes for AMD like ZLUDA, also something called ROCm, I'm just not sure how to integrate that with VoxNovel or PyTorch or however it all works (I'm new)

**Reply by DrewThomasson** [reply]
If you can get torch to detect ROCm as a device then we might be able to get somewhere,

I don't have a AMD device on hand right now to test it is why

**Reply by DrewThomasson** [reply]
ROMc is like AMD's version of CUDA

**Reply by dmarsh400** [reply]
I have 2 machines, one with integrated GPU AMD Ryzen 3 3200g and one with Ryzen 5 5600g / RX 6600 Radeon.

If you have any direction on how to get torch to detect ROCm or at least a starting point I can do some testing if you need me to.

**Reply by DrewThomasson** [reply]
Sadly I haven't had luck getting ROMc running on any computer before

So I'm just as lost as you here 

:/

**Comment by dmarsh400** [comment]
was just looking at PyTorch Lightning at

https://rocm.blogs.amd.com/artificial-intelligence/pytorch-lightning/README.html

**Reply by DrewThomasson** [reply]
hm

**Comment by DrewThomasson** [comment]
## Here's the Deal

Pytorch does support AMD GPUs, which means that xtts potentially supports AMD as well. Pytorch supports AMD, Nvidia, Metal, and CPU devices.

### Your Goal: If you can get this code running on your computer and show me how you did it, I’ll be able to modify VoxNovel to work on your setup. This code is the default and best TTS engine used by VoxNovel.

Get the following script running on your computer with detection of your AMD GPU as the device. You can modify the code as needed.

```python
import torch
from TTS.api import TTS

# Get device
device = "cuda" if torch.cuda.is_available() else "cpu"
print(f"Device selected is: {device}")

# List available TTS models
print(TTS().list_models())

# Initialize TTS
tts = TTS("tts_models/multilingual/multi-dataset/xtts_v2").to(device)

# Run TTS
# Note: Since this is a multilingual voice cloning model, you must set the target speaker WAV and language
# Text to speech - list of amplitude values as output
wav = tts.tts(text="Hello world!", speaker_wav="my/cloning/audio.wav", language="en")

# Text to speech - save to a file
tts.tts_to_file(text="Hello world!", speaker_wav="my/cloning/audio.wav", language="en", file_path="output.wav")
```

This code is based on the [TTS GitHub repository](https://github.com/coqui-ai/TTS). 

For convenience, here is a test `ref.wav` file you can use in place of `"my/cloning/audio.wav"`:

[ref.wav.zip](https://github.com/user-attachments/files/16678510/ref.wav.zip)


I’ll continue to investigate, but I may not be available next week due to school and limited time for side projects. ChatGPT or Claude might also be able to help, but I may not have much time to assist with getting romC running on your computer, especially since I don’t have one on hand at the moment.

:(

**Reply by dmarsh400** [reply]
I will see if I can get it to run and let you know, I will do a little more research into this subject and see what changes need to be made.
